File: Updog.Client/src/post/common/post-info.ts
import { PostType } from './post-type';

/**
 * Information about a post.
 */
export class PostInfo {
    /**
     * Create a new post info.
     * @param id The ID of the post.
     * @param type The type flag of the post.
     * @param title The title of the post.
     * @param body The body (URL or text) of the post.
     * @param author The author's name.
     * @param date The date of posting.
     */
    public constructor(
        public id: number,
        public type: PostType,
        public title: string,
        public body: string,
        public author: string,
        public date: Date
    ) {}
}

File: Updog.Application/Post/Common/PostInfo.cs
using System;
using Updog.Domain;

namespace Updog.Application {
    /// <summary>
    /// Information about a post.
    /// </summary>
    public sealed class PostInfo {
        #region Properties
        /// <summary>
        /// The unqiue ID of the post.
        /// </summary>
        /// <value></value>
        public int Id { get; }

        /// <summary>
        /// Type flag.
        /// </summary>
        public PostType Type { get; }

        /// <summary>
        /// The title of the post.
        /// </summary>
        /// <value></value>
        public string Title { get; }

        /// <summary>
        /// The body of the post. Either a URL or text.
        /// </summary>
        /// <value></value>
        public string Body { get; }

        /// <summary>
        /// The name of the user that created it.
        /// </summary>
        public string Author { get; }

        /// <summary>
        /// When the post was made.
        /// </summary>
        public DateTime Date { get; }
        #endregion

        #region Constructor(s)
        /// <summary>
        /// Create a new post info.
        /// </summary>
        /// <param name="id">The unique ID of the post.</param>
        /// <param name="type">The type of post it is.</param>
        /// <param name="title">The title of the post.</param>
        /// <param name="body">The body of the post.</param>
        /// <param name="author">The author of the post.</param>
        /// <param name="date">The date of posting</param>
        public PostInfo(int id, PostType type, string title, string body, string author, DateTime date) {
            Id = id;
            Type = type;
            Title = title;
            Body = body;
            Author = author;
            Date = date;
        }
        #endregion
    }
}